### ISSUE TITLE
Using testsURL in module declaration rather than a module identifier?

### ISSUE THREAD

**Issue by Ricardh522** [tomwayson/generator-esri-widget#17]
Hello,

I am troubleshooting my widget using this html test page and would like to know why you chose to use the testsUrl variable in the module declaration pointing to the javascript file being tested instead of creating a package in the dojo config on the test page.

Thanks,

Richard

**Comment by tomwayson** [comment]
Honestly, I don't totally recall. I know at one point I had this idea of including a prompt to ask they user if they wanted their tests inline w/ the code, or all under one folder. Looks like maybe I made that change as a step in that direction away from @steveoh's original implementation.

https://github.com/tomwayson/generator-esri-widget/commit/e58b274f11d9da5539f819803caea0d80974279a

**Comment by tomwayson** [comment]
Is that causing problems for you? If you have a suggestion for a better way to do it I'm open to suggestion and/or pull request.

**Comment by Ricardh522** [comment]
If I have any other widgets that I construct during the SubWidget lifecycle, they don't resolve to the correct file location.  Instead they try to point to the esri cdn.

My multiple widgets are primarily for code organization as the project got bigger and for ease in testing.

**Comment by davewilton** [comment]
How are you referencing those sub widgets? Why not use relative paths? This is why I created the below issue but still haven't got around to adding the PR. I don't really see how we can create a package because we don't know what the users package will be called.

https://github.com/tomwayson/generator-esri-widget/issues/13

**Comment by steveoh** [comment]
We kinda took the [shotgun](https://github.com/steveoh/generator-dojo-widget/blob/master/app/templates/_test_page.html#L25) approach to this. If you have your packages defined with the loader then it should work as long as the paths are correct. #13 will only get you a different error message if I'm reading this all correctly.

**Comment by Ricardh522** [comment]
I do use relative paths from my SubWidget to point to 4 other widgets.  When using the cdn these relative paths try to use the cdn url.  That's why I've been using a dojoConfig with packages.  All of my unittests and integration tests are working but I'm having trouble getting my SubWidget to render in the test page.  

Going from a single javascript file widget to multiple widgets has made testing easier and coding more enjoyable; now I just need to figure out how to get it to render like it use to.

**Comment by steveoh** [comment]
[Is this](https://github.com/Ricardh522/WAB-CODE/blob/master/src/widgets/eDoc_Search_Tools/test_page/TestSubWidget.html) the page you are fooling around trying to get your subwidget to work @Ricardh522?

**Comment by Ricardh522** [comment]
Yes, the SubWidget is a WebAppBuilder widget that opens in a border container.  I've had it all developed and demoed using this test page before when it was all in one javascript file.  Now I have refactored the monolith into multiple widgets, wrote unitests, integrations tests, and now am trying to render using the test page.

**Comment by steveoh** [comment]
@Ricardh522 you'll need to define your subwidget package in the [packages config](https://github.com/Ricardh522/WAB-CODE/blob/master/src/widgets/eDoc_Search_Tools/test_page/TestSubWidget.html#L31-L38) similar to [ours](https://github.com/steveoh/generator-dojo-widget/blob/master/app/templates/_test_page.html#L25-L34). I would also suggest that if you are trying to test your subwidget that you **only** test your subwidget and not the entire WAB thingger with your subwidget inside of it.

**Comment by davewilton** [comment]
I'd suggest continuing to use relative paths and change tlmSiblingOfDojo: false in the page config. Then you don't need to worry about a package:

```
var dojoConfig = {
            tlmSiblingofDojo: false
}

```

Alternatively, you can add the "widgets" package to the dojo config and then reference your widgets e.g. "widgets/eDoc_Search_Tools/SubWidget/SubWidget". The WAB knows about the widgets package. e.g:

```
var dojoConfig = {
            packages: [
                {name: "widgets", location: baseurl + "widgets"}
```

**Comment by Ricardh522** [comment]
This is a link to the current TestSubWidget.html file.  It has changed since the one that was linked to in a previous message by @steveoh. [link to file](https://github.com/Ricardh522/WAB-CODE/blob/master/src/widgets/eDoc_Search_Tools/test_page/TestSubWidget.html)
